### PR TITLE
#3 #4 #5 Authentication

### PR DESCRIPTION
**Pull request by archae0pteryx**
#3 #4 #5 

- Add login
- Add logout
- Add basic nav
- Add basic layout component
- Add tests for all

### CODE REVIEW

File: web/src/Routes.tsx [hunk 1]
```diff
@@ -1,17 +1,13 @@
...
-//
-// 'src/pages/HomePage/HomePage.js'         -> HomePage
-// 'src/pages/Admin/BooksPage/BooksPage.js' -> AdminBooksPage
-
 import { Router, Route } from '@redwoodjs/router'
 
 const Routes = () => {
   return (
     <Router>
+      <Route path="/" page={HomePage} name="home" />
```
Comment (craineum) [inline web/src/Routes.tsx:6]: ❓ Should we put this inside a `<Private>` component?

Comment (craineum) [reply web/src/Routes.tsx:6]: ❓ Also the `MainLayout` via the `Set` component like thus: `<Set wrap={MainLayout}>`

**Review by craineum — APPROVED** [review]
Couple of quesitons, but LGTM 🚀

File: web/src/Routes.tsx (hunk 1 above)
Comment (archae0pteryx) [reply web/src/Routes.tsx:6]: @craineum about the private component, I don't think so. The main layout will wrap all of the pages and give it the base style / padding. I don't think it should have logic about permissions.

**Pull request merged**